update-orphaning/index.html

```diff
@@ -181,17 +181,39 @@
       var updateDisabled = data[1]["update-enabled-disabled"]["update-disabled"];
       var first = true;
 
-      // Populate distribution pie chart
-      {
-        var pie = new d3pie("distribution-chart", {
+      // Generates the slices for a d3pie chart.
+      // @param aSlicesArray: an array of arrays in the following format:
+      //                      [[label, value, color],
+      //                       [label, value, color],
+      //                       ...
+      //                      ]
+      // @return An array of d3pie slices that can be used for data.content
+      function generateD3PieSlices(aSlicesArray) {
+        var slices = [];
+        for (s in aSlicesArray) {
```

Comment:
I think you mean to use for..of, not for..in (which will iterate over length as well :(

And you can be clever, if you'd like, and use [destructuring assignment](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) like so:

```javascript
for ([label, value, colour] of aSlicesArray) {
  slices.push({label: label, value: value, color: colour});
}
```

...actually, it might be cleaner still to
```javascript
return aSlicesArray.map(([label, value, colour]) => {
  return {"label": label, "value": value, "color": colour};
});
```